src/client/Microsoft.Identity.Client/Internal/Requests/OnBehalfOfRequest.cs

```diff
@@ -146,7 +151,7 @@ protected override void EnrichTelemetryApiEvent(ApiEvent apiEvent)
 
         protected override KeyValuePair<string, string>? GetCcsHeader(IDictionary<string, string> additionalBodyParameters)
```

Comment:
In regards to this conversation https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/2713#discussion_r653997978

The CCS Method here: https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/e7fd6b8abfdb63dcf2aaa460aeec3616b40ba3ea/src/client/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs#L275

is a virtual method which is overridden by the individual request classes that have the option of returning null when the ccs header cannot be provided, thus skipping all of the additional if's in flows where the CCS header is not available. Also, we now have a new flow in which we can add the CCS header with and it was pretty easy to add it here so I decided to go with this design.